packages/react-dom/src/client/ReactDOMHostConfig.js

```diff
@@ -1004,14 +1005,20 @@ export function didNotHydrateInstance(
   parentProps: Props,
   parentInstance: Instance,
   instance: HydratableInstance,
+  isConcurrentMode: boolean,
 ) {
-  if (__DEV__ && parentProps[SUPPRESS_HYDRATION_WARNING] !== true) {
-    if (instance.nodeType === ELEMENT_NODE) {
-      warnForDeletedHydratableElement(parentInstance, (instance: any));
-    } else if (instance.nodeType === COMMENT_NODE) {
-      // TODO: warnForDeletedHydratableSuspenseBoundary
-    } else {
-      warnForDeletedHydratableText(parentInstance, (instance: any));
+  if (__DEV__) {
+    if (
+      parentProps[SUPPRESS_HYDRATION_WARNING] !== true ||
+      (enableClientRenderFallbackOnHydrationMismatch && isConcurrentMode)
```

Comment:
This case is much simpler so can go first.

We're basically saying that suppressHydrationWarning no longer works to suppress child mismatches and as soon as we remove legacy, the check will go away.